src/dal_select2/views.py

```diff
@@ -106,3 +113,46 @@ def post(self, request):
             'id': text,
             'text': text,
         }))
+
+
+class Select2GroupListView(Select2ListView):
+    def get_item_as_group(self, entry):
+        group = None
+        value = tuple()
+
+        if isinstance(entry, collections.Sequence) and \
+           not isinstance(entry, six.text_type):
+
+            entry_length = len(entry)
+            if(entry_length > 1):
+                group, value = entry[0:2]
+            elif(entry_length > 0):
+                value = entry[0]
+        else:
+            value = (entry, )
+
+        return (group, tuple(value)),
+
+    def get(self, request, *args, **kwargs):
+        """"Return option list with children(s) json response."""
+        results = self.get_list()
+        flat_results = [(group, item) for entry in results
+                        for group, items in self.get_item_as_group(entry)
+                        for item in items]
+
+        if self.q:
+            q = self.convert_case(self.q)
+            flat_results = [(g, x) for g, x in flat_results
+                            if q in self.convert_case(x)]
+        results_dict = {}
+        for group, value in flat_results:
+            results_dict.setdefault(group, [])
+            results_dict[group].append(value)
+
+        return http.HttpResponse(json.dumps({
+            'results':
+                [{"id": x, "text": x} for x in results_dict.pop(None, [])] +
+                [{"id": g, "text": g, "children": [{"id": x, "text": x}
+                                                   for x in l]}
+                 for g, l in six.iteritems(results_dict)]
+        }))
```

Comment:
Hehe that's pretty cool mate, however, we need a test and documentation, otherwise put it in a separate file such as hacks.py so at least we have it for the record until somebody wants to rewrite it in tdd :guitar: